src/DiagramGenerator/Diagram.php

```diff
@@ -150,7 +152,10 @@ public function draw()
             $this->image = $this->image->appendImages(true);
         }
 
-        $this->image->setImageFormat('png');
+        $this->image->setImageFormat($this->board->getImageFormat());
+        if ($this->board->getImageFormat() === Texture::IMAGE_FORMAT_JPG) {
```

Comment:
This should be `$this->image->getImageFormat()`. You want to check instance you're changing, not the one that **should** have the same value (because it might not, for any reason).